### DISCUSSION TITLE
Please implement MTA-STS in the DNS setup

### DISCUSSION THREAD

**Discussion by ghost** [simple-login/app#690] [Feature Requests]
Hello i would like to ask a way to enforce on external domain (our) MTA-STS.
And do the very same between you and our mailbox.
Like that for those who don't use pgp, they can tell only encrypted connexion between server or refuse the mail.

**Comment by c0nfigurati0n** [comment]
Yeah, i think this would be great. Altho the title seems to be a bit miss fitting. But i do recommend they set it up in a way to allow people to use CNAME records for this and that they should then set it up under the following sub "mta-sts.simplelogin.co" (.io also works), after that they should add the following to their dns setup screen "mta-sts.users-custom-domain.com. in CNAME TTL (what-ever) mta-sts.simplelogin.co (or .io)" and then to activate it using "_mta-sts.their-custom-domain.com. in CNAME TTL (what-ever) mta-sts.simplelogin.co (or .io)" and that's it. Now they just have to let their users preform a check and boom, they have mta sts setup on their custom domain. Also i agree they may want to update and set the policy mode to "enforce" and the policy's max‑age to "31,557,600" seconds witch equals to about 1 year (the longer, the better).

**Reply by ghost** [reply]
> Yeah, i think this would be great. Altho the title seems to be a bit miss fitting. But i recommend they do set it up in the like way: Allow people to use CNAME records for this and they should then set it up under the following sub "mta-sts.simplelogin.co" (.io also works), after that you add the following to your guys dns setup screen "mta-sts.users-custom-domain.com. in CNAME TTL (what-ever) mta-sts.simplelogin.co (or .io)" and then to activate it using "_mta-sts.their-custom-domain.com. in CNAME TTL (what-ever) mta-sts.simplelogin.co (or .io)" and that's it. Now they just have to let their users preform a check and boom, they have mta sts setup on their custom domain. Also i agree they may want to update and set the policy mode to "enforce" and the policy's max‑age to "31,557,600" seconds witch equals to about 1 year (the longer, the better).

For the title if you have a better idea i can modify it, it just i do not really know how to say it otherwise.

**Reply by c0nfigurati0n** [reply]
I would change the title to the following "Please implement MTA-STS in the DNS setup".

**Reply by ghost** [reply]
> I would change the title to the following "Please implement MTA-STS in the DNS setup".

Done

**Reply by 0xspade** [reply]
Done implementing all of this, now I'm happy with my simplelogin self-hosted setup, hardenize, and internet.nl score is good, and I also fix some of the spamassassin issues with the help of mail-tester.

I will create a short blog in the future about it how I set up my self-hosted simplelogin, the only thing I need is the updated :D

**Reply by c0nfigurati0n** [reply]
> Done implementing all of this, now I'm happy with my simplelogin self-hosted setup, hardenize, and internet.nl score is good, and I also fix some of the spamassassin issues with the help of mail-tester.
> 
> I will create a short blog in the future about it how I set up my self-hosted simplelogin, the only thing I need is the updated :D

I'm not following! If you set this up already, then good for you. But i am recommending this to the simplelogin team so that everyone can benefit from this when using simplelogin. It's only a matter of time before DANE becomes absolute (i don't hope so). And that's why i gave instructions on how they can easily implement this. @0xspade so your comment doesn't help that much. But it's still great to hear you care about privacy and host a simplelogin instance your self.

**Reply by ghost** [reply]
> Done implementing all of this, now I'm happy with my simplelogin self-hosted setup, hardenize, and internet.nl score is good, and I also fix some of the spamassassin issues with the help of mail-tester.
> 
> I will create a short blog in the future about it how I set up my self-hosted simplelogin, the only thing I need is the updated :D

I not recommend this for many people for... privacy, since using your own self hosted version mean you use your own domain, so it's very easy to link back your domain to you so unanonymize all your email in one go.

**Reply by c0nfigurati0n** [reply]
@BirdInFire that's fair, but the whole point of using SimpleLogin with a custom domain and/or self hosting it is that you have full control over everything. But then again, it depends on your use case and on what you're trying to accomplish.

**Reply by c0nfigurati0n** [reply]
@nguyenkims thoughts?

**Reply by nguyenkims** [reply]
As the MTA-STS is relatively new and the recommended value is still `testing`, I think we would wait a bit before adding it to the doc. Also with TLS being supported on almost all email services now, MTA-STS is rather optional IMO.

**Reply by c0nfigurati0n** [reply]
I mean, TLS, DANE, and DNSSEC, etc are all optional. But they are all used less most of the time. I think MTA-STS is kind of a must, seen as forcing encryption on all emails is better for the security and privacy of your users. But it's understandable you might want to wait a bit on implementing it in a production environment.

**Reply by salixh5** [reply]
@nguyenkims I would argue that just because TLS is supported almost universally, that doesn't make MTA-STS less important. Adoption rate really isn't the issue, the problem is the possibility of MITM attacks.

Because all TLS between email servers is StartTLS, meaning an unencrypted connection is initiated on port 25 and after that the servers will negotiate the start of the TLS tunnel (as you know very well). It would be absolutely trivial for a MITM attacker to just strip the information about StartTLS compatibility from the initial SMTP handshake, nothing crazy required. The email servers then fall back to unencrypted communication, even though they might very well support it.

MTA-STS prevents this scenario because the other mail server immediately knows to NOT proceed if the TLS tunnel cannot be established. Sure, you might also hardcode this information in your code for the biggest email servers and you can already accomplish a lot. But now that there is a way to ascertain this programmatically that works reasonably well, why not take advantage of that? And for your users and their custom domains, you could even offer hosting of the TXT file in an automated fashion (CNAME RR to your server, you get the certificate through LE, yadda yadda), or at least give them a nudge that this is something that they can do to improve security.

It is useful technology, especially *because* TLS encryption between email servers is so widespread.

**Reply by ghost** [reply]
> As the MTA-STS is relatively new and the recommended value is still `testing`, I think we would wait a bit before adding it to the doc. Also with TLS being supported on almost all email services now, MTA-STS is rather optional IMO.

You don't even seem to understand what the "testing" mode is.
it's the way to receive the report without enforcing it... it's used for month to detected potential blocking server (like if it would block gmail).

**Reply by salixh5** [reply]
I mean I haven't seen any best practice to leave MTA-STS on `testing` even after you have tested it for a while and looked after your reports from TLS-RPT, but that doesn't mean there isn't any such opinion that @nguyenkims might be referring to. Probably better to keep the discussion on the topic instead of how much knowledge or understanding about a certain subject anyone might have, as that's not very constructive. As far as I can see, MTA-STS support in the industry is actually pretty far along, so it is fairly safe to use and brings an actual tangible benefit.

**Comment by ghost** [comment]
I hope that native MTA-STS support will arrive at SL (and proton), and that its priority will be reconsidered.
Dane may be better, but MTA-STS, while not common, is still more widely used. Outlook, gmail and yahoo have it, quit not bad. Compatibility is essential here.
Most users won't bother hosting their own MTA-STS policies.

**Reply by c0nfigurati0n** [reply]
Agreed.
